### ISSUE TITLE
docs.microsoft.com refused to connect 

### ISSUE THREAD

**Issue by bybytaren** [pnp/custom-learning-office-365#617]
Example of this issue is on the OOTB playlist "Modern Workplace > Day in your digital life with Office 365 > Start your digital day"
with the asset link to https://docs.microsoft.com/office365/customlearning/dayinthelife?context=context%2Fchromeless&linktarget=true

In my production tenant this asset displays "docs.microsoft.com refused to connect" while in all other tenants it loads properly. 
I read the many issues reported on this error although none describe occurrence on the pre-configured assets. 
The only difference I can think of is that this prod tenant has $optInTelemetry = $false 

Any troubleshooting steps I can be advised would be greatly appreciated

**Comment by v-saikirang** [comment]
Hi @bybytaren,

Thanks for your feedback. Please allow us some time. We will look in to this and get back to you ASAP.

Thanks.

**Comment by amitkh22** [comment]
@bybytaren, Is this issue happening only with this asset "Modern Workplace > Day in your digital life with Office 365 > Start your digital day" or with all the other assets as well which are getting content from “docs.microsoft.com”? If this issue with all the assets(which are showing content from “docs.microsoft.com”), can you please make sure “docs.microsoft.com” is on the allowed list?

**Comment by bybytaren** [comment]
Yes, for all assets at docs.microsoft.com and for none of the assets at support.office.com. 
And yes I have already tried the allowed list
![image](https://user-images.githubusercontent.com/24704255/146543879-1d8b994e-b4c9-486d-bf87-0b2d95e46572.png)

**Comment by amitkh22** [comment]
@bybytaren- Thanks for confirming this. Please allow me sometime while I try to get an answer for this. Meanwhile, can you please share if there are any errors in the browser console when you access that asset? And also, which version of Learning Pathways are you using?

**Comment by bybytaren** [comment]
No errors in the browser and I updated to 4.3.1.0 but the error
still persists

**Comment by amitkh22** [comment]
@bybytaren we are looking into this issue. Due to holiday schedules, please expect a delay in response

**Comment by amitkh22** [comment]
Hi @bybytaren, thank you for being patient. Are you still facing this issue? If yes, most likely the issue is - This link is being blocked by some network security rule for that tenant. The allowed list which you shared earlier only affects if you want to add IFramed content in a text field and this setting doesn't have any effect on Learning Pathways.  Can you please check if you are able to access that link directly outside Learning Pathways?

**Comment by bybytaren** [comment]
Hi, yes the issue is still there, and yes there is no issue for me to access https://docs.microsoft.com/en-us/office365/customlearning/dayinthelife?context=context%2Fchromeless&linktarget=true outside of Learning pathways

**Comment by amitkh22** [comment]
Hi @bybytaren, can you please check and share if you have any errors in the **browser console**. Press F12 once you are on the page and navigate to "Console" tab and copy and share all the errors if you see any.

**Comment by bybytaren** [comment]
`Refused to display 'https://docs.microsoft.com/' in a frame because it set 'X-Frame-Options' to 'sameorigin'.
(index):2762 
        
       crbug/1173575, non-JS module files deprecated.
(anonymous) @ (index):2762
DevTools failed to load source map: Could not load content for https://microsoftgenmab.sharepoint.com/sites/AppCatalog/ClientSideAssets/26c9cd9f-db62-49b1-b102-7f0dad281ea0/en-us.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
DevTools failed to load source map: Could not load content for https://statica.akamai.odsp.cdn.office.net/bld/_layouts/15/16.0.22001.12011/ActivityMonitor.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
DevTools failed to load source map: Could not load content for https://amcdn.msftauth.net/meInlineBackCompat.min.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
custom-learning-web-part_6afa454923dcfaa35ca0.js:1 
        
       Message: Origin https://docs.microsoft.com is not responding in a timely manner to "help_getClientHeight" post message. Default iFrame height set. - AssetView (resizeIFrame)`

**Comment by amitkh22** [comment]
@bybytaren From the error it looks like docs.microsoft.com is not allowed as a safe domain to iframe content in from docs.microsoft.com. Iframed content is treated differently. This could be a setting with your network or with your browser. It isn't a general issue with Learning Pathways as it is working for everyone else. You may want to reach out to your network team to check this out.

**Comment by bybytaren** [comment]
The problem is in the solution or in the tenant.
Same problem occurs on non-company device or on other network.
No problem occurs in other learning pathways installations on other tenants.

**Comment by amitkh22** [comment]
The problem is not with the solution as it is working well in your other tenants as well as for other users also. Let me check something at my end, will update you on the next steps.

**Comment by jrFransen** [comment]
Looking forward to your update @amitkh22. 
We are facing a very similar issue. For us 'https://docs.microsoft.com/' external content source does work, but we are facing an issue with content sources from 3rd parties. Those sites can be accessed directly from a browser, but we get the "refused to connect." error once we open it via M365LP.

**Comment by amitkh22** [comment]
@jrFransen, the possible reason behind your issue is that these URLs doesn't allow content to be iFramed. You can try appending the following query string parameter to the URL, but if that doesn't fix it, you are trying to use one of the pages that is not supported.

?context=context%2Fchromeless&linktarget=true

**Comment by jrFransen** [comment]
Thanks for the fast reply @amitkh22 , unfortunately the errors remains. Now contacting the owner of the site.

**Comment by amitkh22** [comment]
Hi @bybytaren, you need to connect with the people that manage your network firewall and ensure that your machine can access those urls and that they are allowed to be Iframed. It is possible that they have not allowed the iFrames from this domain to be rendered on the SharePoint pages.

**Comment by bybytaren** [comment]
Sorry but maybe I was unclear in my response.
Problem occurs in this tenant and in this learning pathways solution on all
devices (also non-company devices) and from which ever network I access
from.
No problem occurs in my other learning pathways solutions on other tenants,
also when I access from company device and network.

**Comment by dcashpeterson** [comment]
Closed due to inactivity. If this is still an issue please open a new issue and reference this one for context.
